dbms/src/main/java/org/polypheny/db/ddl/DdlManagerImpl.java

```diff
@@ -543,7 +578,7 @@ public void addIndex( CatalogTable catalogTable, String indexMethodName, List<St
         IndexType type = IndexType.MANUAL;
 
         // Make sure that this is a table of type TABLE (and not SOURCE)
-        if ( catalogTable.tableType != TableType.TABLE ) {
+        if ( !(catalogTable.tableType == TableType.TABLE || catalogTable.tableType == TableType.MATERIALIZEDVIEW) ) {
             throw new AlterSourceException();
```

Comment:
Adjust exception for table type view

Reply:
Now using RuntimeException with correct error message.